intake_stac/catalog.py

```diff
@@ -162,13 +162,53 @@ def _get_metadata(self, **kwargs):
         metadata.update(kwargs)
         return metadata
 
+    def stack_bands(self, bands):
+        """
+        Stack the listed bands over the ``band`` dimension.
+
+        Parameters
+        ----------
+        bands : list of strings representing the different bands
+        (e.g. ['B1', B2']).
+
+        Returns
+        -------
+        Catalog entry containing listed bands with ``band`` as a dimension
+        and coordinate.
+
+        """
+        item = {'concat_dim': 'band', 'urlpath': []}
+        titles = []
+        for band in bands:
+            value = self._stac_obj.assets[band]
+            href = value.get('href')
+            type = value.get('type')
+            pattern = href.replace(band, '{band}')
+            if bands.index(band) == 0:
+                item['path_as_pattern'] = pattern
+                item['type'] = type
+            else:
+                if item['type'] != type:
+                    raise ValueError(
+                        'Band stacking failed because bands have different '
+                        'types: {}'.format(', '.join([item['type'], type])))
+                if item['path_as_pattern'] != pattern:
+                    raise ValueError(
+                        'Band stacking failed because band url did not '
+                        'generalize')
```

Comment:
this error could also use a test.

Reply:
This one is harder to test because usually hrefs contain the band key in the url at a predictable location, but this is testing for the case where the key is not at a fixed place in the url. I would have to write a whole new broken catalog to test for that.

Reply:
I did improve the wording though to hopefully make it clearer what is going on.